src/atomate2/common/flows/mpmorph.py

```diff
@@ -0,0 +1,493 @@
+            # fitted volumes, scale appropriately
+            v_ref = vmax if v0 > vmax else vmin
+            eps_0 = (v0 / v_ref) ** (1.0 / 3.0) - 1.0
+            linear_strain = [np.sign(eps_0) * (abs(eps_0) + self.min_strain)]
+
+        deformation_matrices = [np.eye(3) * (1.0 + eps) for eps in linear_strain]
+        deformed_structures = _apply_strain_to_structure(
+            structure, deformation_matrices
+        )
+
+        eos_jobs = []
+        for index in range(len(deformation_matrices)):
+            md_job = self.md_maker.make(
+                structure=deformed_structures[index].final_structure,
+                prev_dir=None,
+            )
+            md_job.name = (
+                f"{self.name} {md_job.name} {len(working_outputs['relax']['volume'])+1}"
+            )
+
+            working_outputs["relax"]["energies"].append(md_job.output.output.energy)
+            working_outputs["relax"]["volume"].append(md_job.output.structure.volume)
+            working_outputs["relax"]["stress"].append(md_job.output.output.stress)
+            eos_jobs.append(md_job)
+
+        recursive = self.make(
+            structure=structure,
+            prev_dir=None,
+            working_outputs=working_outputs,
+        )
+
+        new_eos_flow = Flow([*eos_jobs, recursive], output=recursive.output)
+
+        return Response(replace=new_eos_flow, output=recursive.output)
+
+
+@dataclass
+class MPMorphMDMaker(Maker):
+    """Base MPMorph flow for amorphous solid equilibration.
+
+    This flow uses NVT molecular dynamics to:
+    (1 - optional) Determine the equilibrium volume of an amorphous
+        structure via EOS fit.
+    (2 - optional) Quench the equilibrium volume structure from a higher
+        temperature down to a lower desired "production" temperature.
+    (3) Run a production, longer-time MD run in NVT.
+        The production run can be broken up into smaller steps to
+        ensure the simulation does not hit wall time limits.
+
+    Check atomate2.vasp.flows.mpmorph for MPMorphVaspMDMaker
+
+    Parameters
+    ----------
+    name : str
+        Name of the flows produced by this maker.
+    convergence_md_maker : EquilibrateVolumeMaker
+        MDMaker to generate the equilibrium volumer searcher
+    production_md_maker : Maker
+        MDMaker to generate the production run(s)
+    quench_maker :  SlowQuenchMaker or FastQuenchMaker or None
+        SlowQuenchMaker - MDMaker that quenches structure from
+            high to low temperature
+        FastQuenchMaker - DoubleRelaxMaker + Static that
+            "quenches" structure at 0K
+    """
+
+    name: str = "Base MPMorph MD"
+    convergence_md_maker: Maker | None = None  # check logic on this line
+    # May need to fix next two into ForceFieldMDMakers later..)
+    production_md_maker: Maker | None = None
+    quench_maker: FastQuenchMaker | SlowQuenchMaker | None = None
```

Comment:
maybe add a `__post_init__` to raise early if all (convergence_md_maker production_md_maker quench_maker) are left as `None`. based on the doc string sounds like maybe `production_md_maker` shouldn;t be allowed to be `None` anyway?

Reply:
Added a `__post_init__` to this and the base quench maker classes in the same file